src/Features/Core/Portable/IntroduceUsingStatement/AbstractIntroduceUsingStatementCodeRefactoringProvider.cs

```diff
@@ -46,11 +46,8 @@ public override async Task ComputeRefactoringsAsync(CodeRefactoringContext conte
 
         private async Task<TLocalDeclarationSyntax> FindDisposableLocalDeclaration(Document document, TextSpan selection, CancellationToken cancellationToken)
         {
-            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
-
-            var declarationSyntax =
-                root.FindNode(selection)?.GetAncestorOrThis<TLocalDeclarationSyntax>()
-                ?? root.FindTokenOnLeftOfPosition(selection.End).GetAncestor<TLocalDeclarationSyntax>();
+            var refactoringHelperService = document.GetLanguageService<IRefactoringHelpersService>();
```

Comment:
this approach still force users to put caret on right location for refactoring to be suggested? @mavasani ?

Reply:
If anything this should allow the refactoring for more locations. I was very careful to not to regress (showed before, doesn't show now).